File: 20240528-python-html-converter/html_converter.py
import html as html_converter


class HtmlConverter:

    def convert_to_html(self, filename):
        with open(filename, "r") as f:
            return self.convert_file_to_html(f)

    def convert_file_to_html(self, file):
        html = "<html>"
        for line in file:
            html += "<p>"
            line = line.strip()
            html += self.escape(line, quote=True)
            html += "</p>"
        html += "</html>"
        return html

    def escape(self, line, quote):
        return html_converter.escape(line, quote=quote)


if __name__ == "__main__":
    import sys
    if len(sys.argv) < 2:
        print("Usage: html_converter.py <filename>")
        sys.exit(1)
    filename = sys.argv[1]
    converter = HtmlConverter()
    print(converter.convert_to_html(filename))

File: 20240528-java-html-converter/src/main/java/org/sammancoaching/HtmlConverter.java
package org.sammancoaching;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

import static org.sammancoaching.StringEscapeUtils.escapeHtml;

public class HtmlConverter {

    public String convertToHtml(String filename) throws IOException {
        FileReader in = new FileReader(filename);
        BufferedReader reader = new BufferedReader(in);
        return convertToHtml(reader);
    }

    protected String convertToHtml(BufferedReader reader) throws IOException {
        String line = reader.readLine();
        var html = new StringBuilder("<html>");
        while (line != null) {
            html.append("<p>");
            line = line.trim();
            html.append(escapeHtml(line));
            html.append("</p>");
            line = reader.readLine();
        }
        html.append("</html>");
        return html.toString();
    }

    public static void main(String[] args) throws IOException {
        if (args.length != 1) {
            System.err.println("Usage: java HtmlConverter <filename>");
            System.exit(1);
        }
        HtmlConverter converter = new HtmlConverter();
        System.out.print(converter.convertToHtml(args[0]));
    }
}
